src/main/java/net/sf/jabref/logic/autosaveandbackup/BackupManager.java

```diff
@@ -98,6 +96,13 @@ private void shutdown() {
         }
     }
 
+    static Path getBackupPath(Path originalPath) {
+        Objects.requireNonNull(originalPath.getParent());
+        Path oldFileName = originalPath.getFileName();
+        String newFileName = oldFileName.toString() + BACKUP_FILENAME_ENDING;
```

Comment:
The code was in `FileSaveSession` - I moved it to `FileUtil`.

We don't change the extension, but merely just add a new one. If we changed it, we would have to rely to String magic - at least according to answers to a [similar question on code review](http://codereview.stackexchange.com/questions/110720/adding-old-to-a-file-name/110739#110739?newreg=7a30458926424b07a700b627007e6701).